mne/io/nirx/nirx.py

```diff
@@ -0,0 +1,271 @@
+        #       is currently discarded
+        subject_info = {}
+        names = inf['name'].split()
+        if len(names) > 0:
+            subject_info['first_name'] = \
+                inf['name'].split()[0].replace("\"", "")
+        if len(names) > 1:
+            subject_info['last_name'] = \
+                inf['name'].split()[-1].replace("\"", "")
+        if len(names) > 2:
+            subject_info['middle_name'] = \
+                inf['name'].split()[-2].replace("\"", "")
+        subject_info['birthday'] = inf['age']
+        subject_info['sex'] = inf['gender'].replace("\"", "")
+        # Recode values
+        if subject_info['sex'] in {'M', 'Male', '1'}:
+            subject_info['sex'] = FIFF.FIFFV_SUBJ_SEX_MALE
+        elif subject_info['sex'] in {'F', 'Female', '2'}:
+            subject_info['sex'] = FIFF.FIFFV_SUBJ_SEX_FEMALE
+        # NIRStar does not record an id, or handedness by default
+
+        # Read header file
+        # The header file isn't compliant with the configparser. So all the
+        # text between comments must be removed before passing to parser
+        with open(files['hdr']) as f:
+            hdr_str = f.read()
+        hdr_str = re.sub('#.*?#', '', hdr_str, flags=re.DOTALL)
+        hdr = RawConfigParser()
+        hdr.read_string(hdr_str)
+
+        # Check that the file format version is supported
+        if hdr['GeneralInfo']['NIRStar'] != "\"15.2\"":
+            raise RuntimeError('Only NIRStar version 15.2 is supported')
+
+        # Parse required header fields
+
+        # Extract frequencies of light used by machine
+        fnirs_wavelengths = [int(s) for s in
+                             re.findall(r'(\d+)',
+                             hdr['ImagingParameters']['Wavelengths'])]
+
+        # Extract source-detectors
+        sources = np.asarray([int(s) for s in re.findall(r'(\d+)-\d+:\d+',
+                              hdr['DataStructure']['S-D-Key'])], int)
+        detectors = np.asarray([int(s) for s in re.findall(r'\d+-(\d+):\d+',
+                                hdr['DataStructure']['S-D-Key'])], int)
+
+        # Determine if short channels are present and on which detectors
+        has_short = np.array(hdr['ImagingParameters']['ShortBundles'], int)
+        short_det = [int(s) for s in
+                     re.findall(r'(\d+)',
+                     hdr['ImagingParameters']['ShortDetIndex'])]
+        short_det = np.array(short_det, int)
+
+        # Extract sampling rate
+        samplingrate = float(hdr['ImagingParameters']['SamplingRate'])
+
+        # Read information about probe/montage/optodes
+        # A word on terminology used here:
+        #   Sources produce light
+        #   Detectors measure light
+        #   Sources and detectors are both called optodes
+        #   Each source - detector pair produces a channel
+        #   Channels are defined as the midpoint between source and detector
+        from ...externals.pymatreader import read_mat
```

Comment:
The native format from hardware vendor involves Matlab files ?

Reply:
Unfortunately that is the case.